src/routes/CreateNewSafePage/components/SafeCreationProcess.tsx

```diff
@@ -0,0 +1,240 @@
+import { ReactElement, useState, useEffect, useCallback } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import { backOff } from 'exponential-backoff'
+import { TransactionReceipt } from 'web3-core'
+import { generatePath } from 'react-router'
+import { GenericModal } from '@gnosis.pm/safe-react-components'
+import { makeStyles } from '@material-ui/core/'
+
+import { getSafeDeploymentTransaction } from 'src/logic/contracts/safeContracts'
+import { txMonitor } from 'src/logic/safe/transactions/txMonitor'
+import { userAccountSelector } from 'src/logic/wallets/store/selectors'
+import { FIELD_CREATION_PROXY_SALT } from 'src/routes/open/components/fields'
+import { SafeDeployment } from 'src/routes/opening'
+import { SAFE_ROUTES, WELCOME_ADDRESS } from 'src/routes/routes'
+import { history } from 'src/store'
+import { useAnalytics } from 'src/utils/googleAnalytics'
+import { loadFromStorage, removeFromStorage, saveToStorage } from 'src/utils/storage'
+import { sleep } from 'src/utils/timer'
+import { addOrUpdateSafe } from 'src/logic/safe/store/actions/addOrUpdateSafe'
+import {
+  SAFE_PENDING_CREATION_STORAGE_KEY,
+  CreateSafeFormValues,
+  FIELD_NEW_SAFE_THRESHOLD,
+  FIELD_SAFE_OWNERS_LIST,
+  FIELD_NEW_SAFE_GAS_LIMIT,
+  FIELD_NEW_SAFE_CREATION_TX_HASH,
+  FIELD_CREATE_SUGGESTED_SAFE_NAME,
+  FIELD_CREATE_CUSTOM_SAFE_NAME,
+} from '../fields/createSafeFields'
+import { getSafeInfo } from 'src/logic/safe/utils/safeInformation'
+import { buildSafe } from 'src/logic/safe/store/actions/fetchSafe'
+import { addressBookSafeLoad } from 'src/logic/addressBook/store/actions'
+import { makeAddressBookEntry } from 'src/logic/addressBook/model/addressBook'
+import Paragraph from 'src/components/layout/Paragraph'
+import NetworkLabel from 'src/components/NetworkLabel/NetworkLabel'
+import Button from 'src/components/layout/Button'
+import { boldFont } from 'src/theme/variables'
+
+type ModalDataType = {
+  safeAddress: string
+  safeName?: string
+  safeCreationTxHash?: string
+}
+
+function SafeCreationProcess(): ReactElement {
+  const [safeCreationTxHash, setSafeCreationTxHash] = useState<string | undefined>()
+  const [creationTxPromise, setCreationTxPromise] = useState<Promise<TransactionReceipt>>()
+
+  const { trackEvent } = useAnalytics()
+  const dispatch = useDispatch()
+  const userAddressAccount = useSelector(userAccountSelector)
+
+  const classes = useStyles()
+
+  const [showModal, setShowModal] = useState(false)
+  const [modalData, setModalData] = useState<ModalDataType>({ safeAddress: '' })
+
+  const createNewSafe = useCallback(async () => {
+    const safeCreationFormValues = (await loadFromStorage(SAFE_PENDING_CREATION_STORAGE_KEY)) as CreateSafeFormValues
```

Comment:
If you're casting this, you should null check everything that uses it.

Reply:
I'd say a null check before casting is enough. But yeah, if you check the `loadFromStorage` signature, it can return `undefined`

Reply:
I am not sure why but it seems that its working fine if `undefined` value is returned from `loadFromStorage`:
![Captura de pantalla 2021-09-21 a las 12 10 46](https://user-images.githubusercontent.com/26763673/134153156-e2088923-bd36-4dd9-9911-074645a2f080.png)

and `SafeCreationProcess.tsx` component is not rendered if `safePendingToBeCreated` is undefined

Reply:
A few lines below you retrieve certain values from the object, which could be `undefined`.

Reply:
See line 66. The key wouldn't exist if this is `undefined`.